### PR TITLE
JBEHAVE-1467 Eliminate copy-paste in story reporters

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/reporters/ConcurrentStoryReporter.java [hunk 1]
```diff
@@ -114,311 +115,165 @@ public void storyNotAllowed(Story story, String filter) {
...
+        perform(reporter ->  reporter.restartedStory(story, cause), restartedStory, story, cause);
     }
 
     @Override
     public void storyCancelled(Story story, StoryDuration storyDuration) {
-        crossReferencing.storyCancelled(story, storyDuration);
+        perform(reporter ->  reporter.storyCancelled(story, storyDuration), storyCancelled, story, storyDuration);
+    }
+
+    private void perform(Consumer<StoryReporter> invoke, Method delayed, Object... args) {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/ConcurrentStoryReporter.java]: `invoke` -> `crossReferencingInvoker` ?

Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/ConcurrentStoryReporter.java]: `delayed` -> `delayedMethod` ?

Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/ConcurrentStoryReporter.java]: `args` -> `delayedMethodArgs` ?

File: jbehave-core/src/main/java/org/jbehave/core/reporters/SilentSuccessFilter.java [hunk 2]
```diff
@@ -98,161 +70,104 @@ public void afterStory(boolean givenStory) {
...
             @Override
             public void doNow() {
-                delegate.afterExamples();
+                todoAction.run();
             }
         });
     }
-    
+
+    private State createState(Runnable reportState) {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/SilentSuccessFilter.java]: `reportState` -> `stateReporter`

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Comment by uarlouski** [comment]
In master https://github.com/jbehave/jbehave-core/commit/51ad38d9b5e5ab3b03af064dfa280cdc18bc3dea

**Pull request closed without merging**